packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -1,102 +1,138 @@
 import * as React from 'react';
 import clsx from 'clsx';
 import PropTypes from 'prop-types';
-import { elementTypeAcceptingRef } from '@material-ui/utils';
+import { elementTypeAcceptingRef, deepmerge } from '@material-ui/utils';
 import Collapse from '@material-ui/core/Collapse';
-import { alpha, withStyles } from '@material-ui/core/styles';
+import {
+  alpha,
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
 import { ownerDocument, useForkRef, unsupportedProp } from '@material-ui/core/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import TreeViewContext from '../TreeView/TreeViewContext';
 import { DescendantProvider, useDescendant } from '../TreeView/descendants';
 import TreeItemContent from './TreeItemContent';
+import treeItemClasses, { getTreeItemUtilityClass } from './treeItemClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    listStyle: 'none',
-    margin: 0,
-    padding: 0,
-    outline: 0,
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+    content: ['content'],
+    expanded: ['expanded'],
+    selected: ['selected'],
+    focused: ['focused'],
+    disabled: ['disabled'],
+    iconContainer: ['iconContainer'],
+    label: ['label'],
+    group: ['group'],
+  };
+
+  return composeClasses(slots, getTreeItemUtilityClass, classes);
+};
+
+const TreeItemRoot = experimentalStyled(
+  'li',
+  {},
+  {
+    name: 'MuiTreeItem',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
+  },
+)({
+  listStyle: 'none',
+  margin: 0,
+  padding: 0,
+  outline: 0,
+});
+
+const StyledTreeItemContent = experimentalStyled(
+  TreeItemContent,
+  {},
+  {
+    name: 'MuiTreeItem',
+    slot: 'Content',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+      return deepmerge(
+        {
+          ...(styles.iconContainer && {
+            [`.${treeItemClasses.iconContainer}`]: styles.iconContainer,
```

Comment:
```suggestion
            [`& .${treeItemClasses.iconContainer}`]: styles.iconContainer,
```

Isn't the icon container child of the `TreeItenContent`?